mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+        segment intervals, in the format returned by
+        :func:`mir_eval.io.load_intervals` or
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    labels : list, shape=(n,)
+        reference segment labels, in the format returned by
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    label_set : list
+        An (ordered) list of labels to determine the plotting order.
+        If not provided, the labels will be inferred from existing
+        `yticklabels`.
+        If no `yticklabels` exist, then the sorted set of unique values
+        in `labels` is taken as the label set.
+
+    base : np.ndarray, shape=(n,), optional
+        Vertical positions of each label
+
+    height : np.ndarray, shape=(n,), optional
+        Height for each label
+
+    extend_labels : bool
+        If `False`, only values of `labels` that also exist in `label_set`
+        will be shown.
+
+        If `True`, all labels are shown, with those in `labels` but
+        not in `label_set` appended to the top of the plot.
+        A horizontal line is drawn to indicate the separation between
+        values in or out of `label_set`.
+
+    ax : matplotlib.pyplot.axes
+        An axis handle on which to draw the intervals.
+        If none is provided, a new set of axes is created.
+
+    tick : bool
+        If `True`, sets tick positions and labels on the y-axis.
+
+    kwargs
+        Additional keyword arguments to pass to
+        `matplotlib.patches.Rectangle`.
+
+    Returns
+    -------
+    ax : matplotlib.pyplot.axes._subplots.AxesSubplot
+        A handle to the (possibly constructed) plot axes
+    '''
+
+    if ax is None:
+        # Create a new axis
+        ax = plt.gca()
+
+    if label_set is None:
+        # If we have non-empty pre-existing tick labels, use them
+        label_set = [_.get_text() for _ in ax.get_yticklabels()]
+        if not any(label_set):
+            label_set = []
+    else:
+        label_set = list(label_set)
+
+    # Put additional labels at the end, in order
+    if extend_labels:
+        ticks = label_set + sorted(set(labels) - set(label_set))
+    elif label_set:
+        ticks = label_set
+    else:
+        ticks = sorted(set(labels))
+
+    seg_def_style = dict(linewidth=1)
+
+    seg_map = dict()
+    seg_y = dict()
+
+    style = next(ax._get_patches_for_fill.prop_cycler)
+    if base is None:
+        base = np.arange(len(ticks))
+
+    if height is None:
+        height = np.ones(len(base))
+
+    for y0, yi, lab in zip(base, height, ticks):
+        seg_map[lab] = dict(label=lab)
+        seg_map[lab].update(seg_def_style)
+        seg_map[lab].update(style)
+        seg_map[lab]['facecolor'] = seg_map[lab].pop('color')
+        seg_map[lab].update(kwargs)
+
+        seg_y[lab] = (y0, yi)
+
+    seen = set()
+    for ival, lab in zip(intervals, labels):
+        if lab not in seg_map:
+            continue
+        over_lab = seg_map[lab].get('label', lab)
```

Comment:
When will `'label' not in seg_map[lab]` be `True` (in other words, why are you using `get` here instead of just indexing)?  It seems like it will always be set on line 226, `seg_map[lab] = dict(label=lab)`.